kong/plugins/oauth2/access.lua

```diff
@@ -643,6 +644,15 @@ local function issue_token(conf)
           end
         end
 
+        if not response_params[ERROR] and conf.global_credentials then
+          if kong.plugin.get_id() ~= auth_code.plugin_id then
```

Comment:
How do we handle entries added before the migration?

Reply:
Yeah, that would be a problem. I updated it to become only validate if `auth_code.plugin_id` is present to make the old code pass the validation.